packages/feedback/src/widget/Form.ts

```diff
@@ -0,0 +1,245 @@
+
+  /**
+   * Hides the error message
+   */
+  hideError: () => void;
+
+  /**
+   * Disables the submit button
+   */
+  setSubmitDisabled: () => void;
+
+  /**
+   * Enables the submit button
+   */
+  setSubmitEnabled: () => void;
+}
+
+function retrieveStringValue(formData: FormData, key: string): string {
+  const value = formData.get(key);
+  if (typeof value === 'string') {
+    return value.trim();
+  }
+  return '';
+}
+
+/**
+ * Creates the form element
+ */
+export function Form({
+  options: {
+    showName,
+    showEmail,
+    isAnonymous,
+
+    nameLabel,
+    namePlaceholder,
+    emailLabel,
+    emailPlaceholder,
+    messageLabel,
+    messagePlaceholder,
+    cancelButtonLabel,
+    submitButtonLabel,
+  },
+
+  defaultName,
+  defaultEmail,
+  onCancel,
+  onSubmit,
+}: FormComponentProps): FormComponent {
+  const {
+    el: submitEl,
+    setDisabled: setSubmitDisabled,
+    setEnabled: setSubmitEnabled,
+  } = SubmitButton({
+    label: submitButtonLabel,
+  });
+
+  function handleSubmit(e: Event): void {
+    e.preventDefault();
+
+    if (!(e.target instanceof HTMLFormElement)) {
+      return;
+    }
+
+    try {
+      if (onSubmit) {
+        const formData = new FormData(e.target as HTMLFormElement);
+        const feedback = {
+          name: retrieveStringValue(formData, 'name'),
+          email: retrieveStringValue(formData, 'email'),
+          message: retrieveStringValue(formData, 'message'),
+        };
+
+        onSubmit(feedback);
+      }
+    } catch {
+      // pass
+    }
+  }
+
+  const errorEl = createElement('div', {
+    className: 'form__error-container form__error-container--hidden',
+    ariaHidden: 'true',
+  });
+
+  function showError(message: string): void {
+    errorEl.textContent = message;
+    errorEl.classList.remove('form__error-container--hidden');
+    errorEl.setAttribute('ariaHidden', 'false');
+  }
+
+  function hideError(): void {
+    errorEl.textContent = '';
+    errorEl.classList.add('form__error-container--hidden');
+    errorEl.setAttribute('ariaHidden', 'true');
+  }
+
+  const nameEl = createElement('input', {
+    id: 'name',
+    type: showName ? 'text' : 'hidden',
+    ariaHidden: showName ? 'false' : 'true',
+    name: 'name',
+    className: 'form__input',
+    placeholder: namePlaceholder,
+    value: defaultName,
+  });
+
+  const emailEl = createElement('input', {
+    id: 'email',
+    type: showEmail ? 'text' : 'hidden',
+    ariaHidden: showEmail ? 'false' : 'true',
+    name: 'email',
+    className: 'form__input',
+    placeholder: emailPlaceholder,
+    value: defaultEmail,
+  });
+
+  const messageEl = createElement('textarea', {
+    id: 'message',
+    autoFocus: 'true',
+    rows: '5',
+    name: 'message',
+    className: 'form__input form__input--textarea',
+    placeholder: messagePlaceholder,
+    onKeyup: (e: Event) => {
+      if (!(e.currentTarget instanceof HTMLTextAreaElement)) {
+        return;
+      }
```

Comment:
@mydea wdyt, do we want this runtime check here?

Reply:
I think actually we can maybe remove this altogether. I've read a bunch of times that actually using `disabled` on submit buttons is not recommended for usability/accessibility reasons. Just leave it enabled always and check when submitting if there is no text, and if so, show a corresponding message about _why_ you cannot submit?

Reply:
👍 good point, updated to remove the disabled button